### ISSUE TITLE
`function-signature` should not suggest moving parts of multiline expression body to the same line as function declaration

### ISSUE THREAD

**Issue by 3flex** [pinterest/ktlint#1620]
## Expected Behavior
`function-signature` rule does not flag this code:

```kotlin
private fun asPatterns(rawValue: String?): List<String> =
    rawValue?.trim()
        ?.commaSeparatedPattern(",", ";")
        ?.toList()
        .orEmpty()
```

with message "First line of body expression fits on same line as function signature".

To fix this, the code has to change to:
```kotlin
private fun asPatterns(rawValue: String?): List<String> = rawValue?.trim()
    ?.commaSeparatedPattern(",", ";")
    ?.toList()
    .orEmpty()
```

This moves the first line quite a distance from the wrapped chained calls, making it harder to scan each element of the expression body.

I'd also note that while the [Kotlin coding conventions](https://kotlinlang.org/docs/coding-conventions.html#properties) say

> If the function has an expression body whose first line doesn't fit on the same line as the declaration, put the = sign on the first line and indent the expression body by four spaces.

That doesn't necessarily mean that the inverse is true, i.e. just because the first line of the expression might fit on the same line as the declaration, doesn't mean it should.

I'd suggest that the rule only triggers when the expression body is a single line and would fit on the same line as the function declaration.

## Observed Behavior
See above.

## Steps to Reproduce
Use code similar to above - something that has the expression body starting on a new line followed by wrapped chained calls.

## Your Environment

* Version of ktlint used:
* Relevant parts of the `.editorconfig` settings
* Name and version (or code for custom task) of integration used (Gradle plugin, Maven plugin, command line, custom Gradle task):
* Version of Gradle used (if applicable):
* Operating System and version:

**Comment by paul-dingemans** [comment]
Yeah, I didn't like that part of the coding conventions as well. So, I have added additional (configurable) settings to ktlint. See https://pinterest.github.io/ktlint/rules/configuration-ktlint/#wrapping-the-expression-body-of-a-function and please let me know whether this fits your use case as well.

**Comment by paul-dingemans** [comment]
Closed, due to lack of follow-up.
